src/sqlfluff/rules/L049.py

```diff
@@ -41,79 +42,60 @@ class Rule_L049(Rule_L006):
     """
 
     groups = ("all", "core")
-    # Inherit crawl behaviour from L006
+    crawl_behaviour = SegmentSeekerCrawler({"comparison_operator"})
 
-    def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
         """Relational operators should not be used to check for NULL values."""
         # Context/motivation for this rule:
         # https://news.ycombinator.com/item?id=28772289
         # https://stackoverflow.com/questions/9581745/sql-is-null-and-null
-        if len(context.segment.segments) <= 2:
-            return None  # pragma: no cover
 
         # Allow assignments in SET clauses
-        if context.parent_stack and context.parent_stack[-1].is_type(
+        if context.parent_stack and context.parent_stack[-2].is_type(
             "set_clause_list", "execute_script_statement"
         ):
             return None
 
-        # Allow assignments in EXEC clauses
-        if context.segment.is_type("set_clause_list", "execute_script_statement"):
+        # Allow assignments in EXEC clauses, or any other explicit assignments
+        if context.parent_stack and context.parent_stack[-1].is_type(
+            "set_clause_list", "execute_script_statement", "assignment_operator"
+        ):
+            return None
+
+        # We only care about equality operators.
+        if context.segment.raw not in ("=", "!=", "<>"):
             return None
 
-        segment = FunctionalContext(context).segment
-        # Iterate through children of this segment looking for equals or "not
-        # equals". Once found, check if the next code segment is a NULL literal.
+        # We only care if it's followed by a null.
```

Comment:
Seems a bit clearer...
```suggestion
        # We only care if it's followed by a NULL literal.
```